src/org/opensolaris/opengrok/configuration/messages/StatsMessage.java

```diff
@@ -39,11 +41,38 @@
 
     static final List<String> ALLOWED_OPTIONS = Arrays.asList(new String[]{"get", "reload", "clean"});
 
+    private String statisticsFilePath;
+
+    /**
+     * @return the statisticsFilePath
+     */
+    public String getStatisticsFilePath() {
+        return statisticsFilePath;
+    }
+
+    /**
+     * Sets the statisticsFilePath
+     * @param path a defined value or null
+     */
+    public void setStatisticsFilePath(String path) {
+        this.statisticsFilePath = path;
+    }
+
     @Override
     protected byte[] applyMessage(RuntimeEnvironment env) throws IOException, ParseException {
+        String statsFilePath = getStatisticsFilePath();
+
         if (getText().equalsIgnoreCase("reload")) {
-            env.loadStatistics();
+            if (statsFilePath != null) {
+                env.loadStatistics(new File(statsFilePath));
+                env.getConfiguration().setStatisticsFilePath(statsFilePath);
```

Comment:
Having the `RuntimeEnvironment` singleton is bad by itself when objects query the global state. This actually makes it worse by setting global state values from an object. Setting configuration value is expected in the `config` message however here I'd say it's completely unexpected so it violates the concept of least surprise.